contracts/test/PlonkVerifier.t.sol

```diff
@@ -0,0 +1,432 @@
+        (IPlonkVerifier.PlonkProof memory proof) = abi.decode(result, (IPlonkVerifier.PlonkProof));
+        return proof;
+    }
+
+    /// @dev helper function to generate some dummy but format-valid arguments for
+    /// `prepareOpeningProof` step. The verifyingKey should be fixed/loaded from library,
+    /// proof should be generated via `dummyProof()`, other inputs are from fuzzers.
+    function dummyArgsForOpeningProof(
+        uint64 seed,
+        uint256[] memory publicInput,
+        bytes memory extraTranscriptInitMsg
+    )
+        public
+        returns (
+            IPlonkVerifier.VerifyingKey memory,
+            IPlonkVerifier.PlonkProof memory,
+            V.Challenges memory,
+            Poly.EvalData memory
+        )
+    {
+        IPlonkVerifier.VerifyingKey memory vk = sanitizeVk(VkTest.getVk(), publicInput.length);
+        IPlonkVerifier.PlonkProof memory proof = dummyProof(seed);
+        V.Challenges memory chal =
+            V._computeChallenges(vk, publicInput, proof, extraTranscriptInitMsg);
+
+        Poly.EvalDomain memory domain = Poly.newEvalDomain(vk.domainSize);
+        // pre-compute evaluation data
+        Poly.EvalData memory evalData = Poly.evalDataGen(domain, chal.zeta, publicInput);
+
+        return (vk, proof, chal, evalData);
+    }
+
+    /// Thin wrapper to ensure two G1 points are the same
+    function assertEqG1Point(BN254.G1Point memory a, BN254.G1Point memory b) public {
+        assertEq(a.x, b.x);
+        assertEq(a.y, b.y);
+    }
+}
+
+contract PlonkVerifier_constants_Test is Test {
+    /// @dev Test constants declared matches that from Jellyfish
+    function test_correctConstants() external {
+        string[] memory cmds = new string[](3);
+        cmds[0] = "diff-test";
+        cmds[1] = "plonk-constants";
+
+        bytes memory result = vm.ffi(cmds);
+        (
+            uint256 k1,
+            uint256 k2,
+            uint256 k3,
+            uint256 k4,
+            uint256 betaHX0,
+            uint256 betaHX1,
+            uint256 betaHY0,
+            uint256 betaHY1
+        ) = abi.decode(
+            result, (uint256, uint256, uint256, uint256, uint256, uint256, uint256, uint256)
+        );
+
+        assertEq(k1, V.COSET_K1);
+        assertEq(k2, V.COSET_K2);
+        assertEq(k3, V.COSET_K3);
+        assertEq(k4, V.COSET_K4);
+        assertEq(betaHX0, V.BETA_H_X0);
+        assertEq(betaHX1, V.BETA_H_X1);
+        assertEq(betaHY0, V.BETA_H_Y0);
+        assertEq(betaHY1, V.BETA_H_Y1);
+    }
+}
+
+contract PlonkVerifier_batchVerify_Test is PlonkVerifierCommonTest {
+    /// forge-config: default.fuzz.runs = 30
+    /// @dev Test if some of the user inputs are invalid
+    function testFuzz_revertWhenInvalidArgs(
+        IPlonkVerifier.VerifyingKey[] memory verifyingKeys,
+        uint256[][] memory publicInputs,
+        IPlonkVerifier.PlonkProof[] memory proofs,
+        bytes[] memory extraTranscriptInitMsgs
+    ) external {
+        vm.assume(
```

Comment:
The tests still pass on my machine when I comment this line.

Reply:
this is perfectly normal and expected, `vm.assume` is just a rejection sampling, if the conditions are not met, then they will generate another input, 
in this case, we are trying to test if bad inputs will cause revert. and fuzz-generated inputs (that are not proper proofs and vk and public inputs) are bad most of the time!! so it's normal that all of them revert.